src/pages/index.js

```diff
@@ -117,48 +114,120 @@ function getCardElement(data) {
   const cardLikeButton = cardElement.querySelector(".card__like-btn");
   const cardBinButton = cardElement.querySelector(".card__bin-btn");
 
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-btn_liked");
+  }
+
   cardImageElement.src = data.link;
   cardImageElement.alt = data.alt;
   cardNameElement.textContent = data.name;
 
-  cardLikeButton.addEventListener("click", () => {
-    cardLikeButton.classList.toggle("card__like-btn_liked");
-  });
-
-  cardBinButton.addEventListener("click", () => {
-    cardElement.remove("disabled");
-  });
-
-  cardImageElement.addEventListener("click", () => {
-    openModal(previewModal);
-    previewModalImageElement.src = data.link;
-    previewModalImageElement.alt = data.alt;
-    previewModalCaptionElement.textContent = data.name;
-  });
+  cardLikeButton.addEventListener("click", (evt) => handleLike(evt, data._id));
+  cardBinButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+  cardImageElement.addEventListener("click", () => handleImageClick(data));
 
   return cardElement;
 }
 
 // Event handlers
-function handleEditProfileSubmit(evt) {
+function handleAvatarSubmit(evt) {
   evt.preventDefault();
 
-  profileName.textContent = nameInput.value;
-  profileDescription.textContent = descriptionInput.value;
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileImage.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(setButtonText(submitBtn, false));
```

Comment:
`.finally(setButtonText(submitBtn, false));`

In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


**Please, correct it everywhere in the code**